### PR TITLE
Vishal/09142023

### LINKED ISSUE #36: Log Analytics Workspace resource name max length is 63, not 260
See https://learn.microsoft.com/azure/azure-resource-manager/management/resource-name-rules#microsoftoperationalinsights

### LINKED ISSUE #37: Treat App Insights connection string consistent with instrumentation key

### PR DESCRIPTION
**Pull request by kalalvishal**
Added github Action Workflow.
updated with code required to fix issue #36 and #37.

### CODE REVIEW

File: .github/workflows/bicep-build.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                client-id: ${{ secrets.AZURE_CLIENT_ID }}
+                tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+                subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub create --location westeurope --template-file ./azDeploySecureSub.bicep --verbose
```
Comment (SvenAelterman) [inline .github/workflows/bicep-build.yml:76]: Can't have a hardcoded location

File: .github/workflows/bicep-build.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub validate --location westeurope --template-file ./azDeploySecureSub.bicep
+    
+    # validate the resources before creation
+    validateResources:
```
Comment (SvenAelterman) [inline .github/workflows/bicep-build.yml]: Do we really need different jobs, or are different steps sufficient? We could avoid having to do repetitive checkouts and logins.

File: .github/workflows/bicep-build.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
+## deploy azDeploySecureSub.bicep
+
+name: Azure REDCAP Deployment
```
Comment (SvenAelterman) [inline .github/workflows/bicep-build.yml]: REDCAP should be REDCap

**Review by SvenAelterman — CHANGES_REQUESTED** [review]

File: .github/workflows/bicep-build.yml (hunk 2 above)
Comment (kalalvishal) [reply .github/workflows/bicep-build.yml]: We will keep only two jobs: Validate and Deploy.

File: .github/workflows/bicep-build.yml (hunk 3 above)
Comment (kalalvishal) [reply .github/workflows/bicep-build.yml]: done.

File: .github/workflows/bicep-build.yml (hunk 1 above)
Comment (kalalvishal) [reply .github/workflows/bicep-build.yml:76]: Its fixed now.

Comment (SvenAelterman) [reply .github/workflows/bicep-build.yml:76]: OK, I'll approve this PR but note that this is still hardcoded.

**Review by SvenAelterman — APPROVED** [review]

**Pull request merged**